test/unit/test_quilbase.py

```diff
@@ -159,6 +160,10 @@ def test_eq(self, gate: Gate, name: str, params: List[ParameterDesignator], qubi
         assert not (gate == not_eq_gate)
         assert gate != not_eq_gate
 
+    def test_copy(self, gate: Gate):
+        assert isinstance(copy.copy(gate), Gate)
+        assert isinstance(copy.deepcopy(gate), Gate)
```

Comment:
Would it make sense to also check the output? e.g.
```suggestion
        assert isinstance(copy.deepcopy(gate), Gate)
        assert copy.deepcopy(gate).out() == gate.out()
```

Reply:
Testing something about the contents would be good, I think.

Reply:
It's not really necessary because the important copy logic should be (and is) tested in `quil-py`, and these test suites already serve to test that the transformation between `quil-py` and `pyQuil` is correct. The `__deepcopy__` implementation leverages the same code path for that transformation and probably always should, since separating them is just asking for weird bugs.

Reply:
If we're uncomfortable with that, I'll add them. Not worth bike-shedding over.